plugins/by-name/colorizer/default.nix

```diff
@@ -0,0 +1,251 @@
+{
+  lib,
+  ...
+}:
+let
+  inherit (lib.nixvim) defaultNullOpts;
+  inherit (lib) types;
+in
+lib.nixvim.neovim-plugin.mkNeovimPlugin {
+  name = "colorizer";
+  originalName = "nvim-colorizer.lua";
+  package = "nvim-colorizer-lua";
+
+  maintainers = [ lib.maintainers.GaetanLepage ];
+
+  # TODO introduced 2024-07-06: remove after 25.05
+  imports =
+    let
+      oldPluginPath = [
+        "plugins"
+        "nvim-colorizer"
+      ];
+      newPluginPath = [
+        "plugins"
+        "colorizer"
+      ];
```

Comment:
Actually, just realized... we should move this to lib.deprecation.nix

Reply:
https://github.com/nix-community/nixvim/blob/4b5364a66bffd22536e358214b37068b34287a7a/plugins/deprecation.nix#L20-L23

Reply:
Well, this won't really let me do those internal option renames, will it ?

Reply:
Ah, I suppose you had modernization stuff too. No, that's setup just for top level

Reply:
> Ah, I suppose you had modernization stuff too. No, that's setup just for top level. This is probably cleaner, then. Since you'd have to include this stuff anyways.

Yes, and I would be afraid that it would clash.
What happens if you have both:
- rename `x.a` -> `x.b`
- rename `x` -> `y`

What happens if you set `y.a` ? `x.a` ?

Reply:
> What happens if you have both:
> 
>     * rename `x.a` -> `x.b`
> 
>     * rename `x` -> `y`
> 
> 
> What happens if you set `y.a` ? `x.a` ?

I believe that alias options are smart enough to handle those scenarios, but I'd feel more comfortable seeing it empirically.

Reply:
Yeah, a single entry in `deprecation.nix` may be nicer, but since we have to move all the options to `settings` anyway, it makes little difference.